### ISSUE TITLE
[Feat] Distinguish mapsets by ownership and lock in Data tab

### ISSUE THREAD

**Issue by ninsbl** [OSGeo/grass#714]
**Is your feature request related to a problem? Please describe.**
In multi-user settings the list of mapsets can grow quite a bit, so users may have to scroll a fair bit an pick (and remember) mapsets they technically own and thus can work in.
Unsuccesfully trying to start working in others mapsets can be frustrating, esp. for new beginners.

**Describe the solution you'd like**
In the startup screen it would be nice if mapsets could be
 * filtered (e.g. "owned by me" or with a pattern/autocomplete)
 * displayed differently (e.g. different colors, for "my mapsets", "my locked mapsets", "others mapsets")
This is not purely related to first time users, but for first time users within a GIS team,. So maybe something that could be handled in this years GSoC-project?

**Describe alternatives you've considered**
This is no essential enhancement. So, the alternative is mainly sticking with the current behavior and close the issue as wontfix.

**Additional context**
See earlier discussion:
https://lists.osgeo.org/pipermail/grass-user/2016-February/073665.html

Edit: Changed title from Startup windows to Data tab to better reflect current plans on addressing this.

**Comment by mlennert** [comment]
On 17/06/20 11:04, Stefan Blumentrath wrote:
> *Is your feature request related to a problem? Please describe.*
> In multi-user settings the list of mapsets can grow quite a bit, so 
> users may have to scroll a fair bit an pick (and remember) mapsets they 
> technically own and thus can work in.
> Unsuccesfully trying to start working in others mapsets can be 
> frustrating, esp. for new beginners.
> 
> *Describe the solution you'd like*
> In the startup screen it would be nice if mapsets could be
> 

>   * displayed differently (e.g. different colors, for "my mapsets", "my
>     locked mapsets", "others mapsets")

Isn't this already the case ? It seems to me that these mapsets are 
already shown in a light grey compared to the others in black.

Moritz

**Comment by ninsbl** [comment]
> Isn't this already the case ? It seems to me that these mapsets are already shown in a light grey compared to the others in black. Moritz

Yes, partly. However, own, locked mapsets are shown in the same light grey color as mapsets owned by others. Also, having the option to filter/hide mapsets I do not own would be helpful.

**Comment by petrasovaa** [comment]
This needs to be implemented in data catalog, which will be then used in startup screen. Putting GSoC label on this, although more details need to be discussed how to implement the behavior.

**Comment by wenzeslaus** [comment]
See also #710 and #767 for specifications of the different states.

**Comment by petrasovaa** [comment]
The data catalog should display the inaccessible mapsets with grey color. Additionally, I suggest to provide info for user like that:

`mapsetxxx (active session)`

for mapsets with gislock, alternatively we could use  `(locked)`, not sure what is best

and

`mapsetyyy (user: blabla)`

for mapsets of other users.

This information (lock and ownership) would be stored in the node.data dictionary. The label would be created in DataCatalogNode using label property function.

**Comment by wenzeslaus** [comment]
Now depends on #837.
